services/frontendhost/frontend.go

```diff
@@ -1301,12 +1301,25 @@ func (h *Frontend) GetQueueDepthInfo(ctx thrift.Context, queueRequest *c.GetQueu
 	if _, err = h.prolog(ctx, queueRequest); err != nil {
 		return
 	}
+	var cgDesc *c.ConsumerGroupDescription
 
-	cgUUID := queueRequest.GetKey()
+	cgUUID := queueRequest.GetConsumerGroupName()
 
-	if !common.UUIDRegex.MatchString(cgUUID) { // Special handling to ensure only a UUID is allowed
-		err = &c.BadRequestError{Message: fmt.Sprintf("Consumer group must be given as UUID, not \"%v\"", cgUUID)}
-		return
+	if !common.UUIDRegex.MatchString(cgUUID) {
+		rcgReq := &c.ReadConsumerGroupRequest{
+			ConsumerGroupName: common.StringPtr(queueRequest.GetConsumerGroupName()),
+			DestinationPath:   common.StringPtr(queueRequest.GetDestinationPath()),
+		}
+		cgDesc, err = h.ReadConsumerGroup(ctx, rcgReq)
+		if err != nil {
+			return
+		}
+
+		cgUUID = cgDesc.GetConsumerGroupUUID()
+	} else {
+		if queueRequest.GetDestinationPath() != `` {
```

Comment:
Suggest restructuring this code to make it clear to the reader... How about the following:

var cgUUID string 


if len(request.GetDestinationPath()) > 0 {
 // when dstPth is specified, the name param must be a cg name
   if  isUUID(request.GetConsumerGroupName()) {
      // error out
   }
   cgUUID = lookupUUID(name)
} else {
   // when dstPth is empty, name is *overloaded* as uuid
   cgUUID = request.GetConsumerGroupName()
} 


if !isUUID(cgUUID) {
   // return error
}